### PR TITLE
49 generate pdf

### PR DESCRIPTION
**Pull request by matthopson**

### CODE REVIEW

**Comment by matthopson** [comment]
I think I should move the endpoint that returns the generated PDF from `/api` to `/cases` - thoughts?

**Comment by wbyoko** [comment]
I think it's unlikely this works on DEV. You would also want to model your proxy and lambda off of the `createCourtIssuedOrderPdfFromHtml` workflow to get the headers set correctly.

https://github.com/flexion/ef-cms/blob/develop/shared/src/proxies/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlProxy.js

https://github.com/flexion/ef-cms/blob/develop/web-api/src/courtIssuedOrder/createCourtIssuedOrderPdfFromHtmlLambda.js

File: shared/src/business/useCases/generateDocketRecordPdfInteractor.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
+ * generateDocketRecordPdfInteractor
+ *
+ * @param pdfData {Uint8Array}
```
Comment (kkoskelin) [inline shared/src/business/useCases/generateDocketRecordPdfInteractor.js]: will need to update jsdocs

File: web-client/src/presenter/actions/printDocketRecordAction.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
+import { state } from 'cerebral';
+import printDocketRecordTemplate from '../../views/DocketRecord/printDocketRecordTemplate.html';
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/printDocketRecordAction.js:2]: It's possible the test suite will choke on this.

File: web-client/src/presenter/actions/printDocketRecordAction.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+  const getPartyInfoContent = (caseDetail, helper) => {
+    let partyInfoContent = '';
+
+    const getAddress = contact => {
+      let content = '';
+
+      content += ifTextContent(
+        contact.inCareOf,
+        '<div>c/o' + contact.inCareOf + '</div>',
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/printDocketRecordAction.js:34]: It's certainly not the first (nor last) time we have had to wrangle with creating complete HTML documents with templating and conditionals and so forth.  I feel like there's got to be a better way to do this within our existing framework (without resorting to string concat or `mustache` templates, etc).  This is OK for now, but we should keep our eyes peeled for something better.

**Review by kkoskelin — DISMISSED** [review]
A couple things to address

**Review by wbyoko — APPROVED** [review]

File: web-client/src/presenter/actions/printDocketRecordAction.js (hunk 2 above)
Comment (matthopson) [reply web-client/src/presenter/actions/printDocketRecordAction.js:2]: How / when? Is it intermittent?

File: shared/src/business/useCases/generateDocketRecordPdfInteractor.js (hunk 1 above)
Comment (matthopson) [reply shared/src/business/useCases/generateDocketRecordPdfInteractor.js]: Dah, good catch. I just refactored this.

**Comment by matthopson** [comment]
@kkoskelin If ☝️was not the case, I've got another PR coming up to do some cleanup for this and we can talk through anything else. 😬

**Pull request merged**